clang/lib/StaticAnalyzer/Checkers/StackAddrEscapeChecker.cpp

```diff
@@ -348,12 +357,27 @@ void StackAddrEscapeChecker::checkEndFunction(const ReturnStmt *RS,
 
   ExplodedNode *Node = Ctx.getPredecessor();
 
+  bool ExitingTopFrame =
+      Ctx.getPredecessor()->getLocationContext()->inTopFrame();
+
+  if (ExitingTopFrame && Node->getLocation().getTag() &&
+      Node->getLocation().getTag()->getTagDescription() ==
```

Comment:
Is this a common idiom to check for the tag description? I wonder if we could come up with something that is less stringly typed.

Reply:
Yea, it's not pretty. However, the string comparison only happens if we are leaving the top-level entry point, which happens once per entry point, so performance shouldn't be an issue here.

Factoring out the clean node construction from the ExprEngine seems like a tangential effort with limited benefits.
I'd say let's keep it as-is.

What we should make sure though, is to have a test for this branch of code. That would guard the condition to never get stale. I think we already have tests for this purpose.

Reply:
I've rewritten it with a direct pointer comparison: 532eea2
I think it is not worse and it won't trigger the string-typing thoughts when one reads to code.
Do you agree? I can revert it if it isn't going in the right direction

Reply:
I am happy with the new version.